apimodels/agent_models.go

```diff
@@ -120,7 +120,7 @@ type CreateHost struct {
 	SecurityGroups  []string    `mapstructure:"security_group_ids" json:"security_group_ids" yaml:"security_group_ids" plugin:"expand"`
 	Spot            bool        `mapstructure:"spot" json:"spot" yaml:"spot"`
 	Subnet          string      `mapstructure:"subnet_id" json:"subnet_id" yaml:"subnet_id" plugin:"expand"`
-	UserdataFile    string      `mapstructure:"userdata_file" json:"-" yaml:"-" plugin:"expand"`
+	UserdataFile    string      `mapstructure:"userdata_file" json:"-" yaml:"script" plugin:"expand"`
```

Comment:
Why was this changed to script?

Reply:
So that it is consistent between host.create and the CLI.

Reply:
I'm wary of this change because it could break existing yaml files that use create.host and expect `userdata_file`. This sort of goes along with what I was saying about tags: the data that's included in the file should work for how host.create is expected now; we don't want to make too many allowances/changes for how the CLI works now.

Reply:
Is this still a concern considering that I added the yaml tags (in EVG-7915)? Therefore there shouldn't be an existing use case for someone using yaml with "-" instead of "script". If I don't make this change, then the people who wan't to use the same file to spin up a host from host.create and the CLI won't be able to define a userData script.

Reply:
I guess to "request changes" I need to comment here again rip sorry but yeah, better to be consistent with how host.create works now than with flag names

Reply:
would it work to case on whether the user has provided parameters vs. a file. If they provide parameters we do what we've been doing. If they pass a file we read it into a `CreateHost` and call [CreateHost](https://github.com/evergreen-ci/evergreen/blob/f66132cf5634d302c6ee8afc0666f65258d9ff15/rest/client/methods.go#L700) instead of [CreateSpawnHost](https://github.com/evergreen-ci/evergreen/blob/f66132cf5634d302c6ee8afc0666f65258d9ff15/rest/client/rest.go#L24). This way any file that works for host.create would do the same thing from the CLI as it does for host.create.

Reply:
followed up on slack

Reply:
There already is a separate CLI command to spawn from a file, but the params still get read into the same struct. I can work on implementing "If they pass a file we read it into a CreateHost and call CreateHost instead of CreateSpawnHost" but that would be adding additional work to something that's already working. It's probably doable, but the question is if it's worth it.

Reply:
up to you 😃

Reply:
I don't think it's worth it right now, but it may come up again and be worth it if they end up needing more parameters.

Reply:
I realize now, I think I was wrong when I said "There already is a separate CLI command to spawn from a file". I added a flag, not a separate CLI command." I'll follow up on slack.